Add Travis CI file

bvanderlaan opened the pull request:
Travis CI ([travis-ci.org](https://travis-ci.org)) is a free hosted CI server which supports many
programing languages and platforms. It also has many services including
access to a Cassandra database.

To use Travis CI just define a .travis.yml file which outlines how to setup
your testing environment and how to run tests then allow Travis to monitor
your GitHub repo.

This change set adds a Travis-CI file which will run the test suite against
multiple versions of Node.

**Note**: Travis will run `npm install` and `npm test` by default for node
projects so this file is pretty light. It just defines that we need access to 
a Cassandra database and which version of node to run the tests against.

I picked each LTS version of Node from 0.10 as the `package.json` for this
project states that it can run on any version of Node from 0.10 and up; 
however I see that when run against Node 0.10 there are errors so this 
project probably does not work on Node 0.10 any more.

I have connected my fork to Travis you can see the test runs [here](https://travis-ci.org/bvanderlaan/cassie-odm)

You can see the current status via this badge: [![Build Status](https://travis-ci.org/bvanderlaan/cassie-odm.svg?branch=master)](https://travis-ci.org/bvanderlaan/cassie-odm)

I like using Travis and it will let the project be tested against multiple versions of Node automatically.

bvanderlaan committed:
Add Travis CI file

Travis CI (travis-ci.org) is a free hosted CI server which supports many
programing languages and platforms. It also has many services including
access to a Cassandra database.

To use Travis CI just define a .travis.yaml file which outlines how to setup
your testing environment and how to run tests then allow Travis to monitor
your GitHub repo.

This change set adds a Travis-CI file which will run the test suite against
multiple versions of Node.

Note: Travis will run npm install and npm test by default so this file is
pretty light. It just defines that we need access to a Cassandra database
and which version of node to run the tests against.

I picked each LTS version of Node from 0.10 as the package.json for this
project states that it can run on any version of Node from 0.10 and up.

Flux159 commented on .travis.yml:
```diff
@@ -0,0 +1,20 @@
+services:
+  - cassandra
+node_js:
+  - '0.10'
```
Think its fine to remove 0.10 here; don't know if these numbers are accurate from 2017 but most people should be on v4 or above now: https://nodesource.com/node-by-numbers

I feel like that should also be changed in the package.json as well (would be good to have it in this same diff / pull request)

bvanderlaan replied:
👍 
Node v4 is out of LTS and I think Node v6 is or will be soon. Since everything seems to still work on Node v4 I pushed a change to update the `package.json` to minimum v4

Flux159 reviewed (approved):
Not sure if github is going to force me to re-review after approving, but assuming that you're going to add a commit changing package.json removing 0.10

bvanderlaan committed:
Update minimum Node version to v4

This change set updates the travis CI and package.json file to state that the
minimum supported Node version is v4.

The current code does not work on Node v0.10 any more. It does still work on
v4 so setting that as the minimum for now

bvanderlaan commented:
The failures on Travis have to do with that timeout I added in https://github.com/Flux159/cassie-odm/pull/11

It takes longer then 2sec to drop the keyspace
